stats/views.py

```diff
@@ -311,6 +365,38 @@ def get_context_data(self, **kwargs):
         return context
 
 
+class FragmentTableViewMDS(FragmentTableView):
+    def get_queryset(self):
+        fragment_pks = self.request.session.get('fragment_pks', [])
+        scenario_pk = self.request.session.get('scenario_pk')
+
+        # Include all fragments whose label set matches that of the selected fragment
+        fragment = int(fragment_pks[0])
+        fragment_pks = []
+        scenario = Scenario.objects.get(pk=scenario_pk)
+        tenses = scenario.get_labels()
+        scenario_fragments = scenario.mds_fragments
+        sequence = scenario_fragments.index(fragment)
+        languages = tenses.keys()
+        label_key = tuple(tenses[lang][sequence] for lang in languages)
+
+        for seq, frag in enumerate(scenario_fragments):
+            labels = tuple(tenses[lang][seq] for lang in languages)
+            if labels == label_key:
+                fragment_pks.append(frag)
+
+        target_words = Sentence.objects. \
```

Comment:
From here onwards it's duplicated code from the parent view, consider refactoring.